### PR TITLE
fix exception with ActorProxy

### PR DESCRIPTION
**Pull request by vinayada1**
# Description

Fix exception while creating ActorProxy

#522 

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #522 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: src/Dapr.Actors/Client/ActorProxy.cs [hunk 1]
```diff
@@ -75,7 +75,10 @@ public static TActorInterface Create<TActorInterface>(ActorId actorId, string ac
         /// <returns>Proxy to the actor object.</returns>
         public static object Create(ActorId actorId, Type actorInterfaceType, string actorType)
         {
-            _ = actorInterfaceType as IActor ?? throw new ArgumentException("The interface must implement IActor.", nameof(actorInterfaceType));
+            if(!typeof(IActor).IsAssignableFrom(actorInterfaceType))
```
Comment (rynowak) [inline src/Dapr.Actors/Client/ActorProxy.cs:78]: Fix spacing here

File: src/Dapr.Actors/Client/ActorProxy.cs [hunk 2]
```diff
@@ -75,7 +75,10 @@ public static TActorInterface Create<TActorInterface>(ActorId actorId, string ac
...
+            {
+                throw new ArgumentException("The interface must implement IActor.", nameof(actorInterfaceType));
+            }
```
Comment (rynowak) [inline src/Dapr.Actors/Client/ActorProxy.cs:81]: Unit test? Obviously the success case here was untested

**Review by rynowak — CHANGES_REQUESTED** [review]
Add a unit test

File: src/Dapr.Actors/Client/ActorProxy.cs (hunk 1 above)
Comment (vinayada1) [reply src/Dapr.Actors/Client/ActorProxy.cs]: fixed

File: src/Dapr.Actors/Client/ActorProxy.cs (hunk 2 above)
Comment (vinayada1) [reply src/Dapr.Actors/Client/ActorProxy.cs:81]: added

File: test/Dapr.Actors.Test/ActorProxyTests.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        /// <summary>
+        /// Tests Proxy Creation.
+        /// </summary>
+        [Fact]
+        public void TestCreateActorProxy_Success()
+        {
+            var actorId = new ActorId("abc");
+            var proxy1 = ActorProxy.Create(actorId, "TestActor");
+            Assert.NotNull(proxy1);
```
Comment (rynowak) [inline test/Dapr.Actors.Test/ActorProxyTests.cs:28]: can you please split these up? these are two different tests for two different APIs

**Review by rynowak — APPROVED** [review]

**Pull request merged**